plugins/janus_duktape_extra.c

```diff
@@ -40,7 +198,15 @@ void janus_duktape_register_extra_functions(duk_context *ctx) {
 	if(ctx == NULL)
 		return;
 	JANUS_LOG(LOG_VERB, "Registering extra Duktape functions\n");
+
 	/* Register all extra functions here */
-	duk_push_c_function(ctx, janus_duktape_extra_sample, 0);
-	duk_put_global_string(ctx, "testExtraFunction");
+
+	duk_push_c_function(ctx, http_get, 1);
+	duk_put_global_string(ctx, "http_get");
+
+	duk_push_c_function(ctx, http_post, 2);
+    duk_put_global_string(ctx, "http_post");
+
+    duk_push_c_function(ctx, jwt_validate, 2);
+    duk_put_global_string(ctx, "jwt_validate");
```

Comment:
All these changes don't belong to the `extra` file. They should be in a new *.c/*.h couple, like the existing duktape extensions we incorporated (see the console log stuff, for instance). Since you're adding two separate features (JWT and HTTP), I believe they should be two separate couples. Besides, as I pointed out, the whole jwt and http stuff shoulf be optional, depending on whether the libraries they depend on are available or not. We have plenty of examples in the code on how to check for libcurl presence and support, which you can use as a basis for both.

Reply:
Many thanks for your comments,
We will do the necessary changes and do another push when the code will be ready for it , 
Many thanks...